packages/shared-metrics/src/dependencies.js

```diff
@@ -59,76 +74,198 @@ exports.activate = function activate() {
+        });
+      } else {
+        const fullDirPath = path.join(dependencyDir, dependency);
+        // Only check directories. For example, yarn adds a .yarn-integrity file to /node_modules/ which we need to
+        // exclude, otherwise we get a confusing "Failed to identify version of .yarn-integrity dependency due to:
+        // ENOTDIR: not a directory, open '.../node_modules/.yarn-integrity/package.json'." in the logs.
+        fs.stat(fullDirPath, (statErr, stats) => {
+          if (statErr) {
+            countDownLatch.countDown();
+            logger.warn('Cannot analyse dependency %s due to %s', fullDirPath, statErr.message);
+            return;
+          }
+          if (!stats.isDirectory()) {
+            countDownLatch.countDown();
+            return;
+          }
+
+          addDependency(dependency, fullDirPath, countDownLatch);
+        });
+      }
+    });
+  });
+}
+
+/**
+ * Parses the package.json file in the given directory and then adds the given dependency (with its version) to
+ * preliminaryPayload.
+ *
+ * @param {string} dependency
+ * @param {string} dependencyDirPath
+ * @param {import('./util/CountDownLatch')} countDownLatch
  */
-function addDependency(dependency, packageJsonPath) {
+function addDependency(dependency, dependencyDirPath, countDownLatch) {
+  const packageJsonPath = path.join(dependencyDirPath, 'package.json');
   fs.readFile(packageJsonPath, { encoding: 'utf8' }, (err, contents) => {
     if (err && err.code === 'ENOENT') {
       // This directory does not contain a package json. This happens for example for node_modules/.cache etc.
       // We can simply ignore this.
-      return logger.debug(`No package.json at ${packageJsonPath}, ignoring this directory.`);
+      countDownLatch.countDown();
+      logger.debug(`No package.json at ${packageJsonPath}, ignoring this directory.`);
+      return;
     } else if (err) {
-      return logger.info(
+      countDownLatch.countDown();
+      logger.info(
         'Failed to identify version of %s dependency due to: %s. This means that you will not be ' +
           'able to see details about this dependency within Instana.',
         dependency,
         err.message
       );
+      return;
     }
 
     try {
-      const pckg = JSON.parse(contents);
-      exports.currentPayload[pckg.name] = pckg.version;
-    } catch (subErr) {
+      const parsedPackageJson = JSON.parse(contents);
+      if (!preliminaryPayload[parsedPackageJson.name]) {
+        preliminaryPayload[parsedPackageJson.name] = parsedPackageJson.version;
+      }
+    } catch (parseErr) {
       return logger.info(
         'Failed to identify version of %s dependency due to: %s. This means that you will not be ' +
           'able to see details about this dependency within Instana.',
         dependency,
-        subErr.message
+        parseErr.message
       );
     }
+
+    const potentialNestedNodeModulesFolder = path.join(dependencyDirPath, 'node_modules');
```

Comment:
Also inspect nested node_modules folders recursively (which we failed to do before).